### ISSUE TITLE
Routing logic bug

### ISSUE THREAD

**Issue by jezell** [containous/traefik#924]
When using a host route in a kubernetes ingress controller, traefik doesn't seem to properly handle catch all routes.

Could not get saml requests to fall through like this:

```
- host: www.mysite.com
          http:
             paths:
              - path: /{path:saml.*}
                 backend:
                  serviceName: www-login
                  servicePort: 3000
              - backend:  
                  serviceName: login-client
                  servicePort: 8891
```
Had to get rid of catch all like so to get the saml endpoint to route. The route without any path is getting all the requests.

```
- host: www.mysite.com
          http:
             paths:
              - path: /{path:saml.*}
                backend:
                  serviceName: www-login
                  servicePort: 3000
              - path: /js
                backend:  
                  serviceName: login-client
                  servicePort: 8891
              - path: /images
                backend:  
                  serviceName: login-client
                  servicePort: 8891
              - path: /templates
                backend:  
                  serviceName: login-client
                  servicePort: 8891
              - path: /
                backend:  
                  serviceName: login-client
                  servicePort: 8891
```

In the same ingress controller, we also have a similar configuration with paths and a catch all route at the end, but without the host header rule, and they fall back on catch all route as expected.

**Comment by seryl** [comment]
**Note**: This may require additional fixes from `stripPrefixRegex.go`.

**Comment by ldez** [comment]
Closed by #1638, in the version 1.3.0-rc3.
